### ISSUE TITLE
generator fails if proto self-references

### ISSUE THREAD

**Issue by thetimetraveler** [chrusty/protoc-gen-jsonschema#7]
e.g. 
```
message Foo {
    string name = 1; 
    repeated Foo bar = 2;
}

```
this will cause generate to get stuck in infinite loop until memory limit is hit.

**Comment by wk8** [comment]
Should be fixed by https://github.com/chrusty/protoc-gen-jsonschema/pull/21

**Comment by chrusty** [comment]
@wk8 fixes are in the new release [0.8.0](https://github.com/chrusty/protoc-gen-jsonschema/releases/tag/0.8.0)

Thank you for reporting this, @thetimetraveler
